File: .claude/hooks/websearch-temporal-context.mjs
#!/usr/bin/env node
// Injecte l'année courante dans les requêtes WebSearch sans contexte temporel (PreToolUse)
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';

const TEMPORAL_WORDS = ['latest', 'recent', 'current', 'new', 'now', 'today', 'this year', 'last year'];
const YEAR_PATTERN = /\b20\d{2}\b/;

export function run(input, { currentYear = new Date().getFullYear() } = {}) {
  const query = input.tool_input?.query ?? '';
  if (!query) return null;

  const hasYear = YEAR_PATTERN.test(query);
  const hasTemporal = TEMPORAL_WORDS.some((w) => query.toLowerCase().includes(w));

  if (hasYear || hasTemporal) return null;

  return {
    hookSpecificOutput: {
      hookEventName: 'PreToolUse',
      modifiedToolInput: { query: `${query} ${currentYear}` },
    },
  };
}

/* v8 ignore next 5 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  const result = run(input);
  if (result) process.stdout.write(JSON.stringify(result));
}

File: .claude/hooks/websearch-temporal-context.py
#!/usr/bin/env python3
# @hookstack pre-websearch-temporal-context
"""Injects the current year into temporal-ambiguous WebSearch queries (PreToolUse)."""
import json
import re
import sys
from datetime import datetime

TEMPORAL_WORDS = [
    "latest",
    "recent",
    "current",
    "new",
    "now",
    "today",
    "this year",
    "last year",
]
YEAR_PATTERN = re.compile(r"\b20\d{2}\b")


def run(input_data, *, current_year=None):
    if current_year is None:
        current_year = datetime.now().year
    query = (input_data.get("tool_input") or {}).get("query") or ""
    if not query:
        return None

    has_year = bool(YEAR_PATTERN.search(query))
    has_temporal = any(w in query.lower() for w in TEMPORAL_WORDS)
    if has_year or has_temporal:
        return None

    return {
        "hookSpecificOutput": {
            "hookEventName": "PreToolUse",
            "modifiedToolInput": {"query": f"{query} {current_year}"},
        }
    }


if __name__ == "__main__":
    data = json.load(sys.stdin)
    result = run(data)
    if result:
        sys.stdout.write(json.dumps(result))
